sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -2150,7 +2150,9 @@ private async Task RunProducerManagementAsync(CancellationToken cancellationToke
                             if ((!cancellationToken.IsCancellationRequested)
                                 && (_activePartitionStateMap.TryGetValue(partition, out var partitionState))
                                 && (partitionState.BufferedEventCount > 0)
-                                && (partitionState.PartitionGuard.Wait(PartitionPublishingGuardAcquireLimitMilliseconds, cancellationToken)))
+                                // Do a quick synchronous check first
+                                && (partitionState.PartitionGuard.Wait(0, cancellationToken) || await partitionState.PartitionGuard.WaitAsync(PartitionPublishingGuardAcquireLimitMilliseconds, cancellationToken)
+                                    .ConfigureAwait((false))))
```

Comment:
```suggestion
                                && ((partitionState.PartitionGuard.Wait(0, cancellationToken)) || (await partitionState.PartitionGuard.WaitAsync(PartitionPublishingGuardAcquireLimitMilliseconds, cancellationToken).ConfigureAwait((false)))))
```

nit:  A couple of minor formatting suggestions, to align with surrounding code.